lxd/storage/backend_lxd.go

```diff
@@ -2387,6 +2387,22 @@ func (b *lxdBackend) CreateCustomVolumeFromCopy(projectName string, volName stri
 		return fmt.Errorf("Failed to negotiate copy migration type: %v", err)
 	}
 
+	// If we're copying block volumes, the target block volume needs to be
+	// at least the size of the source volume, otherwise we'll run into
+	// "no space left on device".
+	var volSize int64
+
+	if contentType == drivers.ContentTypeBlock {
```

Comment:
@monstermunchkin I'm coming into this a bit cold, but is there a reason you couldn't use the same approach we used for block VM root disks here? https://github.com/lxc/lxd/blob/master/lxd/storage/backend_lxd.go#L702-L708

I.e. using drivers.BlockDiskSizeBytes(rootDrivePath)?

Reply:
That decreases the diff quite a bit ;)